### ISSUE TITLE
Tags not Being Applied with Terraform

### ISSUE THREAD

**Issue by smslattery** [localstack/localstack#3193]
# Type of request: This is a ...

[x] bug report
[ ] feature request

# Detailed description

When creating certain resources by running `terraform apply`, any tags associated with the resource are not being applied.

When running `terraform apply` a second time, the tags show up as expected.

This has been observed for:
- aws_network_acl
- aws_route_table
- aws_internet_gateway
- aws_eip

## Expected behavior

Output from terraform apply shows tags applied successfully (using aws_network_acl as example case):
```
Outputs:

nacl_id = acl-36560435
nacl_tags = {
  "name" = "some-nacl"
}
subnet_id = subnet-5d57fded
subnet_tags = {
  "name" = "some-subnet"
}
vpc_id = vpc-1de1ccb6
vpc_tags = {
  "name" = "some-vpc"
}
```

## Actual behavior

Output from terraform apply shows empty map (using aws_network_acl as example case):
```
Outputs:

nacl_id = acl-36560435
nacl_tags = {}
subnet_id = subnet-5d57fded
subnet_tags = {
  "name" = "some-subnet"
}
vpc_id = vpc-1de1ccb6
vpc_tags = {
  "name" = "some-vpc"
}
```

# Steps to reproduce

## Command used to start LocalStack

`IMAGE_NAME=localstack/localstack-full:0.12.1 localstack start`

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)

This was executed using terraform v0.12.6

```
$ terraform apply -auto-approve
aws_vpc.localstack_vpc: Creating...
aws_vpc.localstack_vpc: Creation complete after 1s [id=vpc-1de1ccb6]
aws_subnet.localstack_vpc_subnet: Creating...
aws_subnet.localstack_vpc_subnet: Creation complete after 0s [id=subnet-5d57fded]
aws_network_acl.localstack_nacl: Creating...
aws_network_acl.localstack_nacl: Creation complete after 0s [id=acl-36560435]

Apply complete! Resources: 3 added, 0 changed, 0 destroyed.

Outputs:

nacl_id = acl-36560435
nacl_tags = {}
subnet_id = subnet-5d57fded
subnet_tags = {
  "name" = "some-subnet"
}
vpc_id = vpc-1de1ccb6
vpc_tags = {
  "name" = "some-vpc"
}

$ terraform apply -auto-approve
aws_vpc.localstack_vpc: Refreshing state... [id=vpc-1de1ccb6]
aws_subnet.localstack_vpc_subnet: Refreshing state... [id=subnet-5d57fded]
aws_network_acl.localstack_nacl: Refreshing state... [id=acl-36560435]
aws_network_acl.localstack_nacl: Modifying... [id=acl-36560435]
aws_network_acl.localstack_nacl: Modifications complete after 0s [id=acl-36560435]

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.

Outputs:

nacl_id = acl-36560435
nacl_tags = {
  "name" = "some-nacl"
}
subnet_id = subnet-5d57fded
subnet_tags = {
  "name" = "some-subnet"
}
vpc_id = vpc-1de1ccb6
vpc_tags = {
  "name" = "some-vpc"
}
```

## Terraform Config Files

main.tf
```
resource "aws_vpc" "localstack_vpc" {
  cidr_block                       = "10.0.0.0/16"
  tags = {
    name = "some-vpc"
  }
}

resource "aws_subnet" "localstack_vpc_subnet" {
  vpc_id     = aws_vpc.localstack_vpc.id
  cidr_block = aws_vpc.localstack_vpc.cidr_block
  tags = {
    name = "some-subnet"
  }
}

resource "aws_network_acl" "localstack_nacl" {
  vpc_id     = aws_vpc.localstack_vpc.id
  subnet_ids = [aws_subnet.localstack_vpc_subnet.id]

  tags = {
    name = "some-nacl"
  }
}
```

site.tf
```
provider "aws" {
  region                      = "us-east-1"
  access_key                  = "mock_access_key"
  secret_key                  = "mock_secret_key"
  s3_force_path_style         = true
  skip_credentials_validation = true
  skip_metadata_api_check     = true
  skip_requesting_account_id  = true
  endpoints {
    ec2            = "http://localhost:4566"
  }
}
```

outputs.tf
```
output "vpc_id" {
  value       = aws_vpc.localstack_vpc.id
}

output "vpc_tags" {
  value       = aws_vpc.localstack_vpc.tags
}

output "subnet_id" {
  value       = aws_subnet.localstack_vpc_subnet.id
}

output "subnet_tags" {
  value       = aws_subnet.localstack_vpc_subnet.tags
}

output "nacl_id" {
  value       = aws_network_acl.localstack_nacl.id
}

output "nacl_tags" {
  value       = aws_network_acl.localstack_nacl.tags
}
```

**Comment by smslattery** [comment]
This seems like the same thing I encountered in https://github.com/localstack/localstack/issues/3031
With the 0.12.1 version, it no longer gets the `Error: error waiting for EC2 VPC (vpc-fefd8ae2) IPv6 CIDR to become disassociated: couldn't find resource (2 retries)` and instead succeeds on the second try.

**Comment by whummer** [comment]
Thanks for reporting @smslattery . This should be fixed upstream in spulec/moto#3430 and is now available in LocalStack - can you please pull the `latest` Docker image and give it another try? Please report here if the problem persists. Thanks

**Comment by smslattery** [comment]
Tested and confirmed for:

- aws_network_acl
- aws_route_table
- aws_internet_gateway
